File: array/max-consecutive-ones.py
# https://leetcode.com/problems/max-consecutive-ones/
# tags - array
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
#         count = 0
#         max_count = 0
#         num_digits = len(nums)
#         for i in range(num_digits):
#             if nums[i] == 0:
#                 continue
            
#             # We've seen a 1. Increment the count.
#             count += 1
#             if i+1 == num_digits or nums[i+1] == 0:
#                 max_count = max(max_count, count)
#                 # if count > max_count:
#                 #     max_count = count
#                 count = 0
#         return max_count
        
        last_zero_index = 0
        max_ones_count = 0
        for i, bit in enumerate([0] + nums + [0]):
            if bit == 0:
                max_ones_count = max(max_ones_count, i - last_zero_index - 1)
                last_zero_index = i
        return max_ones_count
            

File: array/max-consecutive-ones.js
// https://leetcode.com/problems/max-consecutive-ones/
// tags - array
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function (nums) {
  let maxCount = 0;
  for (let i = 0, count = 0; i < nums.length; i++) {
    if (nums[i] === 1) {
      count++;
      if (i === nums.length - 1 || nums[i + 1] === 0) {
        maxCount = Math.max(maxCount, count);
        count = 0;
      }
    }
  }
  return maxCount;
};
